src/subscription_manager/gui/registergui.py

```diff
@@ -137,6 +137,51 @@ def pop(self, index=None):
         p = self._list.pop()
         return p
 
+class ServerInfo(object):
+    """Object holding server connection information
+
+    A container class useful for holding the connection information
+    retreived via the screens.
+    """
+    def __init__(self,
+                 hostname=None,
+                 port=None,
+                 prefix=None,
+                 proxy_hostname=None,
+                 proxy_port=None,
+                 proxy_username=None,
+                 proxy_password=None):
+        self.hostname = hostname
+        self.port = port
+        self.prefix = prefix
+        self.proxy_hostname = proxy_hostname
+        self.proxy_port = proxy_port
+        self.proxy_username = proxy_username
+        self.proxy_password = proxy_password
+
+    @staticmethod
+    def from_config(config):
```

Comment:
custom constructors should be @classmethod

```
@classmethod
def from_config(cls, config):
    return cls(hostname=<blah blah blah>)
```

or

```
@classmethod\
def from_config(cls, config):
    server_info = cls()
    cls.hostname = config.get('server', 'hostname')
   <.. etc ..>
```
And called  as ServerInfo.from_config(CFG) 

The latter has the advantage where the __init__ potentially doesn't need to be passed all the args, if they
are going to get created with reasonable empty defaults.

Though, for this case, I think it would be easier to just use a plain python dict for server_info.
I don't think the class buys us much yet.